fastlane/Fastfile

```diff
@@ -879,54 +878,80 @@ def simulator_version
 # Example:
-#   get_test_accounts_state
-#     => "TFFFF"
-def get_test_accounts_state
-  uri = URI.parse(get_required_env("UI_TESTS_ACCOUNTS_URL"))
+#   get_test_accounts_hash
+#     => {"0"=>"4079", "1"=>"free", "2"=>"free", "3"=>"free"}
+def get_test_accounts_hash
+  uri = URI.parse(get_required_env("UI_TESTS_ACCOUNTS_JSON_URL"))
   response = Net::HTTP.get_response(uri)
   accounts_state = response.body.chomp
-  UI.message("State of test accounts: #{accounts_state}")
-  accounts_state
+  JSON.parse(accounts_state)
 end
 
-# Returns the index of first "F" (Free) occurence in the test
-# accounts keyvalue, or crashes fastlane if it was not found.
-# Example:
-#   find_free_test_account
-#     => 2
+# Finds the index of first free account in accounts hash, and marks it as accupied by current build.
+# Crashed fastlane if free account was not found.
 def find_free_test_account
-  $used_test_account_index = get_test_accounts_state.index("F")
+  accounts_hash = get_test_accounts_hash
+  UI.message("Looking for a free test account...")
+  UI.message("Accounts state: #{accounts_hash}")
+
+  $used_test_account_index = accounts_hash.key('free')
+
   UI.user_error!("Could not find free UI Test account. Quitting.") if $used_test_account_index.nil? 
   UI.message("Free account index: #{$used_test_account_index}")
+  change_test_account_availability("#{get_required_env("CIRCLE_BUILD_NUM")}")
 end
 
-# Replaces a character at `index` with `availability` value.
+# Replace a value in a key which is equal to $used_test_account_index global variable
 # Example:
-#   change_test_account_availability("F")
-#     => FFFFF
+#   change_test_account_availability("free")
+#     => {"0"=>"free", "1"=>"free", "2"=>"free", "3"=>"free"}
+#
+#   change_test_account_availability("4820")
+#     => {"0"=>"4820", "1"=>"free", "2"=>"free", "3"=>"free"}
 def change_test_account_availability(availability)
-  accounts_state = get_test_accounts_state
-  accounts_state[$used_test_account_index] = availability
-  UI.message("Writing new account state: #{accounts_state}")
-  uri = URI.parse(get_required_env("UI_TESTS_ACCOUNTS_URL"))
+  accounts_hash = get_test_accounts_hash
+  accounts_hash[$used_test_account_index.to_s] = availability
+  submit_accounts_hash(accounts_hash)
+end
+
+def sanitize_test_accounts
+  accounts_hash = get_test_accounts_hash
+  UI.message("Sanitizing stale test accounts...")
+  UI.message("Accounts before sanitizing: #{accounts_hash}")
+
+  accounts_hash.each do |key, value|
+    next if value == "free" || is_job_running(value)
+    accounts_hash[key] = "free"
+    submit_accounts_hash(accounts_hash)
+  end  
+end
+
+def is_job_running(job_number)
+  uri = URI.parse("https://circleci.com/api/v2/project/gh/Automattic/simplenote-ios/job/#{job_number}")
+  response = Net::HTTP.get_response(uri)
+  return if response.code != "200"
+
+  response_hash = JSON.parse(response.body)
+  puts "Status of Job Number #{job_number} is #{response_hash["status"]}"
```

Comment:
Maybe this `puts` was a debugging feature, but I think it might be useful to keep it in here. It could become:

```suggestion
  UI.message("Status of CircleCI Job Number #{job_number} is #{response_hash['status']}")
```

Or maybe `verbose` if we want to print it only when a developer is looking for extra information.

```suggestion
  UI.verbose("Status of CircleCI Job Number #{job_number} is #{response_hash['status']}")
```

---

Given the logs are already "noisy" and that if a CI job fails it's helpful to have as much information about why in the log for that job itself, because re-running the job might result into a different behavior, I'd say let's use `message`.

Reply:
Yes, the `puts` slipped in here from the local file where I was testing this. I intended to keep the message, but did not change it to `UI.message`. Thank you!